### PR TITLE
Phase -2, Milestone-3 Mapping

### PR DESCRIPTION
**Pull request by shashanknarang27**

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs [hunk 1]
```diff
@@ -232,10 +254,14 @@ public async Task CheckOnePatient(string templateName, string samplePath, DataTy
         {
             var result = await ConvertData(templateName, samplePath, dataType);
             var patients = result.SelectTokens("$.entry[?(@.resource.resourceType == 'Patient')].resource.id");
-            if (templateName == "ADT_A40" || templateName == "ADT_A41")
+            if (templateName == "ADT_A40" || templateName == "ADT_A41" || templateName == "ADT_A45" || templateName == "ADT_A47")
```
Comment (yankunhuang-pku) [inline src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs:257]: If the special templates here will increase all the time, can we consider using list to store them?

Comment (shashanknarang27) [reply src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs:257]: ok sure, will store templates which are having no patient resource in a list. I will update it and will push the changes

Comment (shashanknarang27) [reply src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs]: @yankunhuang-pku : We have updated the RuleBasedTests file to contain the templates name in list for both non and multiple patient resource

File: src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs [hunk 2]
```diff
@@ -511,6 +538,16 @@ internal static class ResourceFilter
                 "source",
             };
 
+            public static readonly List<string> _noPatientTemplate = new List<string>
```
Comment (BoyaWu10) [inline src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs:541]: Can we make this field private?

File: src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs [hunk 3]
```diff
@@ -511,6 +538,16 @@ internal static class ResourceFilter
...
+            {
+                "ADT_A40", "ADT_A41", "ADT_A45", "ADT_A47",
+            };
+
+            public static readonly List<string> _multiplePatientTemplate = new List<string>
```
Comment (BoyaWu10) [inline src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs:546]: Same for this one.

File: src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs (hunk 2 above)
Comment (shashanknarang27) [reply src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs:541]: Sure

File: src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs (hunk 3 above)
Comment (shashanknarang27) [reply src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs:546]: Sure , I will make it private and push the changes

**Comment by yankunhuang-pku** [comment]
Could you help to resolve conflicts? @shashanknarang27

**Comment by shashanknarang27** [comment]
> Could you help to resolve conflicts? @shashanknarang27

Sure, I am working on it

**Comment by shashanknarang27** [comment]
@yankunhuang-pku : We have resolved the conflicts.

File: src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs [hunk 4]
```diff
@@ -526,6 +563,18 @@ internal static class ResourceFilter
                 (string input) => DateTime.TryParse(input, out _),
                 (string input) => bool.TryParse(input, out _),
             };
+
+            public static readonly List<Func<string, bool>> NonPatientTemplates = new List<Func<string, bool>>
+            {
+                // Templates that dosn't contains patient resource
```
Comment (BoyaWu10) [inline src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs:569]: Nit: typo `don't contain`

File: src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs [hunk 5]
```diff
@@ -526,6 +563,18 @@ internal static class ResourceFilter
...
+                _noPatientTemplate.Contains,
+            };
+
+            public static readonly List<Func<string, bool>> MultiplePatientTemplates = new List<Func<string, bool>>
+            {
+                // Templates that contains multiple patient resource
```
Comment (BoyaWu10) [inline src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs:575]: Nit: typo `contain multiple patient resources`

File: src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs (hunk 4 above)
Comment (shashanknarang27) [reply src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs]: Fixed

File: src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs (hunk 5 above)
Comment (shashanknarang27) [reply src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs]: @BoyaWu10 : Fixed the same

**Review by BoyaWu10 — APPROVED** [review]

**Review by yankunhuang-pku — APPROVED** [review]

**Pull request merged**